### PR TITLE
[ADD] base_user_gravatar

### PR DESCRIPTION
**Pull request by Endika**
Use you Gravatar image in to your Odoo

### CODE REVIEW

File: res_users_with_gravatar/README.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+along with this program. If not, see <http://www.gnu.org/licenses/agpl-3.0-standalone.html>.
+
+
+Credits
+=======
+
+Contributors
+------------
+
+* Endika Iglesias <endika2@gmail.com>
```
Comment (eLBati) [inline res_users_with_gravatar/README.rst]: Please follow this template https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: res_users_with_gravatar/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    'version': '1',
+    'author': 'Endika Iglesias',
+    'category': 'Tools',
+    'website': 'http://www.endikaiglesias.com',
+    'depends': ['base'],
+    'data': [
+        'views/res_users_view.xml',
+    ],
+    'installable': True,
+}
```
Comment (eLBati) [inline res_users_with_gravatar/__openerp__.py]: Also, please follow this template https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py

**Comment by sebalix** [comment]
Hi,
I think the module should be renamed to `base_user_gravatar` or something like that to be consistent with others modules. Someone propose a better name?

**Comment by pedrobaeza** [comment]
That name LGTM

**Comment by Endika** [comment]
Ok @sebalix and @pedrobaeza

File: base_user_gravatar/README.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+=====
+
+To configure this module, you need to:
+
+* go to user profile
+* pulse "Get Gravatar image" button
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{repo_id}/{branch}
```
Comment (StefanRijnhart) [inline base_user_gravatar/README.rst]: That would be https://runbot.odoo-community.org/runbot/149/8.0 in this case.

File: base_user_gravatar/models/res_users.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    @api.one
+    def get_gravatar_image(self):
+        email = str(self.email) or ''
+        fail_example = self._get_gravatar_base64('fail@email.gravatar')
+        user_gravatar = self._get_gravatar_base64(email)
+        if fail_example != user_gravatar:
+            self.write({'image': user_gravatar})
+        else:
+            raise Warning(
+                _("You don't have Gravatar image to this %s email." % (
```
Comment (StefanRijnhart) [inline base_user_gravatar/models/res_users.py]: Something like 'There is no Gravatar image for this email (%s)'

File: base_user_gravatar/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# © 2015 Endika Iglesias
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Synchronize Gravatar image',
+    'version': '1',
```
Comment (StefanRijnhart) [inline base_user_gravatar/__openerp__.py]: I think as per convention, the version should now be 8.0.1.0.0

**Comment by StefanRijnhart** [comment]
@Endika, thanks. About the CLA, I see that your CLA has been signed and your Github handle is in the OCA database correctly. Maybe someone added it just now. In any case, it is alright.

**Comment by Endika** [comment]
Ok @StefanRijnhart

File: base_user_gravatar/README.rst [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+Use you Gravatar image in to your Odoo
+
+Usage
+=====
+
+To configure this module, you need to:
+
+* go to user profile
+* pulse "Get Gravatar image" button
```
Comment (dreispt) [inline base_user_gravatar/README.rst]: s/pulse/press

**Comment by dreispt** [comment]
minor remark, but :+1:

**Comment by Endika** [comment]
@dreispt Ok

File: base_user_gravatar/models/res_users.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+# © 2015 Endika Iglesias
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+from openerp.exceptions import Warning
+from openerp.tools.translate import _
+import base64
+import hashlib
+import urllib2
+
```
Comment (moylop260) [inline base_user_gravatar/models/res_users.py:8]: Executing `isort res_users.py` the order import is:

``` python
 import base64
 import hashlib
 import urllib2

from openerp import api, models
from openerp.exceptions import Warning
from openerp.tools.translate import _
```

More info [here](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports)

**Comment by moylop260** [comment]
Small comment but :+1: (without test it)

**Comment by moylop260** [comment]
Check beta messages:

``` bash
Start lint check just in modules changed
************* Module base_user_gravatar.models.res_users
base_user_gravatar/models/res_users.py:6: [R8101(openerp-exception-warning), ] Import `Warning` should be renamed as UserError `from openerp.exceptions import Warning as UserError`
Found 1 errors in modules changed.
```

More info [here](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms)

**Comment by Endika** [comment]
Ok, @moylop260

**Comment by Endika** [comment]
Rebase commit .

**Comment by StefanRijnhart** [comment]
Thanks @endika for the follow ups! Failing travis is unrelated to this change (bar the inclusion of additional tests). But have you got any idea why your rebase now shows existing commits and changes in the diff?

**Comment by Endika** [comment]
@StefanRijnhart  ok, solved

**Comment by StefanRijnhart** [comment]
Thanks again! :+1:

**Comment by IObregon** [comment]
:+1:

File: base_user_gravatar/views/res_users_view.xml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+            <field name="inherit_id" ref="base.view_users_form"/>
+            <field name="arch" type="xml">
+                <field name="login" position="after">
+                    <button name="get_gravatar_image" type="object" string="Get Gravatar image" class="oe_edit_only"/>
+                </field>
+            </field>
+        </record>
+
+    </data>
+</openerp>
```
Comment (tafaRU) [inline base_user_gravatar/views/res_users_view.xml]: No newline at the end of file.

**Comment by tafaRU** [comment]
A little nitpicking, but :+1:

Thanks!

**Comment by Endika** [comment]
@tafaRU Thx

**Comment by moylop260** [comment]
Travis error of other modules.
Then merge

**Comment by Endika** [comment]
thx @moylop260

**Pull request merged**